pkg/gatewayserver/io/ws/format.go

```diff
@@ -46,5 +46,11 @@ type Formatter interface {
 	// This function optionally returns a byte stream to be sent as response to the upstream message.
 	HandleUp(ctx context.Context, raw []byte, ids ttnpb.GatewayIdentifiers, conn *io.Connection, receivedAt time.Time) ([]byte, error)
 	// FromDownlink generates a downlink byte stream that can be sent over the WS connection.
-	FromDownlink(ctx context.Context, uid string, down ttnpb.DownlinkMessage, bandID string, concentratorTime scheduling.ConcentratorTime, dlTime time.Time) ([]byte, error)
+	FromDownlink(ctx context.Context, down ttnpb.DownlinkMessage, bandID string, concentratorTime scheduling.ConcentratorTime, dlTime time.Time) ([]byte, error)
+}
+
+// TimeSyncer abstracts away time transferring activities to websocket based gateways.
+type TimeSyncer interface {
+	// TransferTime generates a spurious time transfer message for a particular server time.
+	TransferTime(ctx context.Context, t time.Time, conn *io.Connection) ([]byte, error)
```

Comment:
Why is this a separate interface? This is just another downstream message right?

Reply:
In order to avoid having to implementing it in other websocket frontends which do not support it. There is no point in creating a timer if the format does not support it.